jwst/photom/photom.py

```diff
@@ -553,8 +549,13 @@ def calc_miri(self, ftab):
                 mid_time = self.input.meta.exposure.mid_time
                 correction = miri_mrs.time_correction(self.input, self.detector,
                                                       ftab, mid_time)
+                inv_correction_sq = 1 / (correction * correction)
                 self.input.data /= correction
                 self.input.err /= correction
+                self.input.var_poisson *= inv_correction_sq
+                self.input.var_rnoise *= inv_correction_sq
+                if self.input.var_flat is not None and np.size(self.input.var_flat) > 0:
```

Comment:
This check is already done above for sens2d but given https://github.com/spacetelescope/jwst/pull/9320/files#r2014792315 , I reverted stuff and repeated the whole check here though maybe eventually this check can be done only once.